test/mocha/unit/rpc/SingleJsonRpcProvider.test.ts

```diff
@@ -72,4 +87,24 @@ describe('SingleJsonRpcProvider', () => {
     expect(blockNumber).equals(123456)
     expect(spy.calledOnce).to.be.true
   })
+
+  it('maybeSyncHealthScore', async () => {
+    provider['enableDbSync'] = true
+    const DB_HEALTH_SCORE = -1000
+    const stubSyncer = sandbox.createStubInstance(HealthStateSyncer)
+    stubSyncer.maybeSyncHealthScoreWithDb.returns(Promise.resolve({ synced: true, healthScore: DB_HEALTH_SCORE }))
+    provider['healthStateSyncer'] = stubSyncer
+
+    const getBlockNumber = sandbox.stub(SingleJsonRpcProvider.prototype, '_getBlockNumber' as any)
+    getBlockNumber.resolves(123456)
+
+    const blockNumber = await provider.getBlockNumber()
+    expect(blockNumber).equals(123456)
+
+    // Wait to make sure all callbacks are executed.
```

Comment:
Honestly I think it should be fine. The worst case is that we start serving a request before finishing syncing with DB so we failed this request. We cannot do `await` in constructor, and we cannot block `return` in `SingleJsonRpcProvider.wrappedFunctionCall` to avoid adding unnecessary latencies to the user. So I think keep the health state syncing under the desk is acceptable.

Reply:
I think it's worth adding a comment on the fire and forget behavior, I didn't pick up on that initially

Reply:
Yes, added comment for doing "fire and forget".
Also I think it confuses people if I use `.then() + .catch()` because it does make it harder to follow. Now I changed them into `await + try/catch`. Hopefully it's much easier to read.